### PR TITLE
create template for cherry-pick requests

### PR DESCRIPTION
**Pull request by itsjustkevin**
Add a template for cherry-pick requests.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.md:5]: Add a label to filter the requests more easily. Then subscribe to the label to keep an eye on how this template is working out https://dart-github-label-notifier.firebaseapp.com/  (internal link).

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.md:10]: I was going to say it's probably easier to link to the GitHub URL for the PR, but it's a nice captcha for whether people should be filing these.  Not to gatekeep, but if you don't know what a commit sha is you probably won't be able to fill out the rest of this successfully.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.md:12]: I don't think we can usually answer this question because we don't know what what parent it will be cherry-picked onto.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.md:14]: The issue is probably already described in the linked GitHub issue.  More relevant for a cherry-pick would be questions about the customer impact.
- Approximately who will hit this issue (ex. all Flutter devs, Windows developers, all end-customers, apps using X framework feature)?
- What is the impact (ex. visual jank on Samsung phones, app crash, cannot ship an iOS app)? Does it impact development (ex. `flutter doctor` crashes when Android Studio is installed), or the shipping production app (the app crashes on launch)
- Is there a workaround?

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.md:18]: This answer would be covered by the customer impact-type questions.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.md:22]: Also add "Link to PR(s) to cherry-pick"

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.md:21]: Would be interesting to have a "test coverage" question: Are you confident your fix is well-tested by automated tests.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.md:23]: May also be good to have an optional "Steps to manually validate fix" if possible to manually test on the hotfix branch (it's not always possible).  It's still amazing to me that we don't do that--we just cherry-pick it and if the tests pass usually no one does a manual smoke test to see if the general product still works, or if the bug is fixed.

**Review by jmagman — COMMENTED** [review]
I bet you're going to get all kinds of wild barebones bug-not-cherry-pick reports because people will think it will escalate their bug being fixed in a hotfix. You might want to give the triagers a heads up that they should close any issue that doesn't reasonably fill out this template, and maybe a script for them as to why it's being closed.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.md:22]: Never mind the SHA question covers it.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (itsjustkevin) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.md:5]: I've created two new labels:

1. cp: review
2. cp: approved

This could solve the problem around contributors having to specify which release their cherry pick applies to.  I have a low concern for contributors abusing the cherry pick system as they do not utilize the current cp: labels often.  If this becomes an issue, we can pivot.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (itsjustkevin) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.md:10]: Agreed.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (itsjustkevin) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.md:12]: This space is a little jumbled.  The current imagined flow is:

1. User wants to submit a cherry-pick request.
2. User fills out form.
3. Release manager notes the candidate branch in the issue comments.
4. User checks for clean merge.
     -  `git checkout <candidate-branch>` 
     -  `git cherry-pick <commit-hash>`
5. User notes if there are merge conflicts or not

Maybe there is a world where we can write an automated comment to the issue that gives merge instructions and notes the correct candidate branch.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.md]: > contributors abusing the cherry pick system as they do not utilize the current cp: labels

Random people don't have permissions to add labels, but they can create new issues.  I'm not worried about #flutter-hackers abusing this.

**Comment by Hixie** [comment]
LGTM modulo Jenn's comments; I'll let her do the final review. :-)

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.md]: It can't be a required field then or you would never be able to get past 2.  I don't think it really matters if it's a clean merge or not.  If it's not a clean merge then it means the author or reviewer of the PR may need to help the cherry-picker fix up the merge conflicts, but I don't think that should be a deciding factor to whether a cherry-pick request should be accepted.  With or without merge conflicts you can still bork things badly, it still needs to be tested.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (jmagman) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.md]: I don't think this is "community", which I think of as Flutter contributor and developers, but often end-users of the app who don't know or care that the app is using Flutter.  I said "customer" because it either impacts developers, who are our customers, or end-users, who are the developer's customers.  I'm not wed to the word though, the point is "what is the impact and who does it impact". 

If this doesn't get fixed, how bad is it?  Did the developer's app get corrupted? Does the app crash on launch? Does it only crash on the latest version of Android?  Can the developer simply work around this issue, like by upgrading their IDE? etc.

Overall I think the questions a cherry-picker needs for approval are:
1. What bug does this fix?
2. What are the pros (impact, to whom)?
3. What are the cons (risk)?

File: .github/ISSUE_TEMPLATE/7_cherry_pick.md
Comment (itsjustkevin) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.md]: Agree on removing this field.

I still think it is valuable to have the author of the PR test that it merges cleanly.  

I see a near future state where there will be a rotation of people releasing the Flutter SDK.  For a process that already takes days to complete, not having to wait for a response and investigation from the PR authors during the release window makes the release easier and more predictable.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.yml
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.yml:18]: Note this may be multiple commits.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.yml
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.yml:35]:
```suggestion
      placeholder: Android apps cannot compile without this fix.
```

File: .github/ISSUE_TEMPLATE/7_cherry_pick.yml
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.yml:43]: Maybe?
```suggestion
      placeholder: Relaunch the app
```

File: .github/ISSUE_TEMPLATE/7_cherry_pick.yml
Comment (jmagman) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.yml:72]: Or imply we want something more detailed
```suggestion
      placeholder: Step 1. `flutter run` the app on Android...
```

**Review by jmagman — APPROVED** [review]
@godofredoc and @christopherfujino may have opinions as well since they have handled plenty of hotfixes.

LGTM

**Review by christopherfujino — APPROVED** [review]
lgtm

File: .github/ISSUE_TEMPLATE/7_cherry_pick.yml
Comment (godofredoc) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.yml:52]: A link to how to calculate the risk may be useful.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.yml
Comment (godofredoc) [inline .github/ISSUE_TEMPLATE/7_cherry_pick.yml:63]: Maybe a link to the tests in the PR? developers may be tempted to say yes even without checking

File: .github/ISSUE_TEMPLATE/7_cherry_pick.yml
Comment (itsjustkevin) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.yml:52]: Do we have a link like this already? The [cherry-pick wiki page](https://github.com/flutter/flutter/wiki/Flutter-Cherrypick-Process) has a few points.  Maybe we can collaborate on putting together a criteria for this.

File: .github/ISSUE_TEMPLATE/7_cherry_pick.yml
Comment (itsjustkevin) [reply .github/ISSUE_TEMPLATE/7_cherry_pick.yml:63]: The tests will show up in the PR, I don't personally think adding a link will deter this behavior.

**Pull request closed without merging**
